test/library/packages/HDF5/HDF5Preprocessors.chpl

```diff
@@ -0,0 +1,62 @@
+module HDF5Preprocessors {
+  use HDF5_HL;
+
+  class AddNPreprocessor: HDF5Preprocessor {
+    const n: int;
+
+    proc preprocess(A: []) {
+      forall a in A {
+        a += 1;
+      }
+    }
+  }
+
+  class ScriptPreprocessor: HDF5Preprocessor {
+    const script: string;
+
+    proc preprocess(A: []) {
+      use FileSystem, Path, Spawn;
+
+      try! {
+        // opentmp() doesn't seem to give me a file I can get the name of :(
+        //var f = opentmp();
+        //const scriptName = f.realPath();
+
+        const scriptName = "hdf5TempScript.bash";
+        var f = open(scriptName, iomode.cw);
+
+        // write the script to a file
+        {
+          var writer = f.writer();
+          writer.writeln(script);
```

Comment:
Would it be worth having a way to give `ScriptPreprocessor` the name of an existing script, instead of the contents?  That way you don't need write access to the current directory, and there's no race between tasks opening (and truncating) and writing the same data to the same file while tasks that got there earlier are trying to execute it.